test/controllers/search_controller_test.rb

```diff
@@ -321,6 +321,38 @@ class SearchControllerTest < ActionDispatch::IntegrationTest
     end
   end
 
+  test 'advanced search form retains values with spaces' do
+    VCR.use_cassette('advanced all spaces',
+                     allow_playback_repeats: true,
+                     match_requests_on: %i[method uri body]) do
+      query = {
+        q: 'some data',
+        citation: 'a citation',
+        contributors: 'some contribs',
+        fundingInformation: 'a fund',
+        identifiers: 'some ids',
+        locations: 'some locs',
+        subjects: 'some subs',
+        title: 'a title',
+        source: 'katsu sauce',
+        advanced: 'true'
+      }.to_query
+      get "/results?#{query}"
+      assert_response :success
+      assert_nil flash[:error]
+
+      assert_select 'li', 'Keyword anywhere: some data'
+      assert_select 'li', 'Citation: a citation'
+      assert_select 'li', 'Contributors: some contribs'
+      assert_select 'li', 'Funders: a fund'
+      assert_select 'li', 'Identifiers: some ids'
+      assert_select 'li', 'Locations: some locs'
+      assert_select 'li', 'Subjects: some subs'
+      assert_select 'li', 'Title: a title'
+      assert_select 'li', 'Source: katsu sauce'
```

Comment:
I don't think this set of asserts actually tests the fix as I believe would pass on the main branch. To test that the form inputs are populated we'll need to select from something more like `input#advanced-identifiers` etc. (that syntax not tested but something like that... I think.

Reply:
Ah, you're absolutely right. Thanks for catching that -- will fix.